compiler/rustc_lint/src/lib.rs

```diff
@@ -39,6 +39,7 @@
 #![feature(min_specialization)]
 #![feature(never_type)]
 #![feature(rustc_attrs)]
+#![feature(trait_upcasting)]
```

Comment:
Is the feature stable enough to use?

Reply:
Seems it's headed for stabilization. See https://github.com/rust-lang/rust/issues/65991#issuecomment-1670127881 and https://github.com/rust-lang/rust/issues/65991#issuecomment-1817552398. FCP ended yesterday.

Reply:
Yes, and also, trait_upcasting is used elsewhere in the compiler if I recall correctly